gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcCallContext.java

```diff
@@ -310,6 +311,12 @@ public Integer getChannelAffinity() {
     return this.extraHeaders;
   }
 
+  @Override
+  public RetryingContext getRetryContext() {
+    // TODO: transfer relevant state from the ApiCallContext
+    return RetryingContext.newBuilder().build();
```

Comment:
It is called `getRetryContext()`, not `createRetryContext()`. Also, I believe it should really be `getRetryContext()`, taking into account that essentially `ApiCallContext` can be used directly there is a 1:1 relationship between the objects (here it  becomes 1:m). I think retrying context should be a field and passed to the constructor for all the coyping operations (like it is already done for `channel`, `callOptions` and others).

Reply:
`RetryContext` will contain totalTimeout and tracer. As it stands my intention was for these fields to live on `ApiContext` and to be copied on `RetryContext` when needed. I think we might have different visions for this.

In the future, when a user wants to set the total timeout, I was expecting the user to do something like:
```java
GrpcCallContext.createDefault()
   .withTotalTimeout(Duration.ofSeconds(2))
```

If I understand your vision, it would be something like:

```java
GrpcCallContext.createDefault()
  .withRetryContext(
    RetryContext.newBuilder()
       .withTotalTimeout(Duration.ofSeconds(2))
       .build()))
```

Which feels a bit clunky.

Please let me know if I understand correctly and if you think this where you want to take this.